aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config_test.go

```diff
@@ -0,0 +1,94 @@
+package aviatrix
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAviatrixControllerBgpMaxAsLimitConfig_basic(t *testing.T) {
+	skipAcc := os.Getenv("SKIP_CONTROLLER_BGP_MAX_AS_LIMIT_CONFIG")
+	if skipAcc == "yes" {
+		t.Skip("Skipping Controller BGP Max AS Limit Config test as SKIP_CONTROLLER_BGP_MAX_AS_LIMIT_CONFIG is set")
+	}
+	msgCommon := ". Set SKIP_CONTROLLER_BGP_MAX_AS_LIMIT_CONFIG to yes to skip Controller BGP Max AS Limit Config tests"
+	resourceName := "aviatrix_controller_bgp_max_as_limit_config.test_bgp_max_as_limit"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			preAccountCheck(t, msgCommon)
+		},
+		Providers:    testAccProvidersVersionValidation,
+		CheckDestroy: testAccCheckControllerBgpMaxAsLimitConfigDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccControllerBgpMaxAsLimitConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckControllerBgpMaxAsLimitConfigExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "max_as_limit", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccControllerBgpMaxAsLimitConfigBasic() string {
+	return `
+resource "aviatrix_controller_bgp_max_as_limit_config" "test_bgp_max_as_limit" {
+	max_as_limit = 1
+}
+`
+}
+
+func testAccCheckControllerBgpMaxAsLimitConfigExists(n string) resource.TestCheckFunc {
```

Comment:
This function needs to actually call `client.GetControllerBgpMaxAsLimit(context.Background())` and verify that it is configured